homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,573 @@
+<specVersion>
+<major>1</major>
+<minor>0</minor>
+</specVersion>
+<URLBase>http://{0}:{1}/</URLBase>
+<device>
+<deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
+<friendlyName>Home Assistant Emulated Hue Bridge ({0})</friendlyName>
+<manufacturer>Royal Philips Electronics</manufacturer>
+<manufacturerURL>http://www.philips.com</manufacturerURL>
+<modelDescription>Philips hue Personal Wireless Lighting</modelDescription>
+<modelName>Philips hue bridge 2015</modelName>
+<modelNumber>BSB002</modelNumber>
+<modelURL>http://www.meethue.com</modelURL>
+<serialNumber>1234</serialNumber>
+<UDN>uuid:2f402f80-da50-11e1-9b23-001788255acc</UDN>
+</device>
+</root>
+"""
+
+        resp_text = xml_template.format(
+            self.config.host_ip_addr, self.config.listen_port)
+
+        return self.Response(resp_text, mimetype='text/xml')
+
+
+class HueUsernameView(HomeAssistantView):
+    """Handle requests to create a username for the emulated hue bridge."""
+
+    url = '/api'
+    name = 'hue:api'
+    requires_auth = False
+
+    def __init__(self, hass):
+        """Initializ the instance of the view."""
+        super().__init__(hass)
+
+    def post(self, request):
+        """Handl a POST request."""
+        data = request.json
+
+        if 'devicetype' not in data:
+            return self.Response("devicetype not specified", status=400)
+
+        json_response = [{'success': {'username': '12345678901234567890'}}]
+
+        return self.json(json_response)
+
+
+class HueLightsView(HomeAssistantView):
+    """Handle requests for getting and setting info about entities."""
+
+    url = '/api/<username>/lights'
+    name = 'api:username:lights'
+    extra_urls = ['/api/<username>/lights/<entity_id>',
+                  '/api/<username>/lights/<entity_id>/state']
+    requires_auth = False
+
+    def __init__(self, hass, config):
+        """Initialize the instance of the view."""
+        super().__init__(hass)
+        self.config = config
+        self.cached_states = {}
+
+    def get(self, request, username, entity_id=None):
+        """Handle a GET request."""
+        if entity_id is None:
+            return self.get_lights_list()
+
+        if not request.base_url.endswith('state'):
+            return self.get_light_state(entity_id)
+
+        return self.Response("Method not allowed", status=405)
+
+    def put(self, request, username, entity_id=None):
+        """Handle a PUT request."""
+        if entity_id is None:
+            return self.Response("Not found", status=404)
+
+        if request.base_url.endswith('state'):
+            content_type = request.environ.get('CONTENT_TYPE', '')
+            if content_type == 'application/x-www-form-urlencoded':
+                # Alexa send JSON data with a form data content type, for
+                # whatever reason. and Werkzeug parses form data automatically,
+                # so we need to do some gymnastics to get the JSON we need

```

Comment:
I'm not sure. The Hue API docs don't specify the content type, but since the data is JSON, I'd imagine it's implied to be "application/json". I don't own a real bridge, otherwise I'd see what the Hue smartphone app does.